### ISSUE TITLE
Component blueprint --path option breaks addons with component blueprints that use the built-in blueprint

### ISSUE THREAD

**Issue by trabus** [ember-cli/ember-cli#4183]
As reported [here](https://github.com/Vestorly/ember-cli-emblem/issues/20).

Basically, the ember-cli-emblem addon has a component blueprint that does a lookup on the built-in component and calls locals with the command options, like so:
```
locals: function(options) {
    return this.lookupBlueprint('component').locals(options);
}
```

Because the `--path` option default is not properly being set, it comes through as undefined, breaking the `ember g component` command for ember-cli-emblem, and possibly more addons that make use of the same practice.

I've got a PR started for the fix (which is extremely simple), and starting on the test, but I don't see any existing acceptance tests where we're setting up the tests in a way that I need. Basically, I've got to generate an ember app, install a fixture addon that overrides the component blueprint, and then generates a component.

**Comment by buschtoens** [comment]
This is really weird. I get the same error again.

```
jan@ubuntu:~/code/veltro$ ember v
version: 0.2.7
Could not find watchman, falling back to NodeWatcher for file system events.
Visit http://www.ember-cli.com/#watchman for more info.
node: 2.3.0
npm: 2.11.0

jan@ubuntu:~/code/veltro$ ember g component application/nav-bar
version: 0.2.7
Could not find watchman, falling back to NodeWatcher for file system events.
Visit http://www.ember-cli.com/#watchman for more info.
installing
Arguments to path.join must be strings
TypeError: Arguments to path.join must be strings
    at Object.posix.join (path.js:467:13)
    at __path__ (/home/jan/code/veltro/node_modules/ember-cli/blueprints/component/index.js:26:23)
    at /home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:629:66
    at Array.map (native)
    at Class.Blueprint.generateFileMap (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:629:37)
    at Class.Blueprint._locals (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:815:22)
    at Class.Blueprint._process (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:459:22)
    at Class.Blueprint.install (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:492:15)
    at Class.module.exports.Task.extend.run (/home/jan/code/veltro/node_modules/ember-cli/lib/tasks/generate-from-blueprint.js:52:49)
    at Class.module.exports.Command.extend.run (/home/jan/code/veltro/node_modules/ember-cli/lib/commands/generate.js:76:17)
jan@ubuntu:~/code/veltro$ ember g component application-navbar
version: 0.2.7
Could not find watchman, falling back to NodeWatcher for file system events.
Visit http://www.ember-cli.com/#watchman for more info.
installing
Arguments to path.join must be strings
TypeError: Arguments to path.join must be strings
    at Object.posix.join (path.js:467:13)
    at __path__ (/home/jan/code/veltro/node_modules/ember-cli/blueprints/component/index.js:26:23)
    at /home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:629:66
    at Array.map (native)
    at Class.Blueprint.generateFileMap (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:629:37)
    at Class.Blueprint._locals (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:815:22)
    at Class.Blueprint._process (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:459:22)
    at Class.Blueprint.install (/home/jan/code/veltro/node_modules/ember-cli/lib/models/blueprint.js:492:15)
    at Class.module.exports.Task.extend.run (/home/jan/code/veltro/node_modules/ember-cli/lib/tasks/generate-from-blueprint.js:52:49)
    at Class.module.exports.Command.extend.run (/home/jan/code/veltro/node_modules/ember-cli/lib/commands/generate.js:76:17)
````

Removing ember-cli-emblem fixes this.

**Comment by buschtoens** [comment]
Stupid me. The fix didn't land in time for the 0.2.7 release. I guess `master` works.
